### PR TITLE
fix `Trainer.train(resume_from_checkpoint=False)` is causing an exception

### PR DESCRIPTION
**Pull request by PhilipMay**
fix with regression test for #12970

### CODE REVIEW

**Comment by PhilipMay** [comment]
All Tests green and ready for review. 👍

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1005,6 +1005,8 @@ def train(
             kwargs:
                 Additional keyword arguments used to hide deprecated arguments
         """
+        # fix for https://github.com/huggingface/transformers/issues/12970
```
Comment (sgugger) [inline src/transformers/trainer.py:1008]: No need to put it in the code.

File: tests/test_trainer.py [hunk 2]
```diff
@@ -827,6 +827,26 @@ def test_resume_training_with_randomness(self):
         self.assertAlmostEqual(a, a1, delta=1e-8)
         self.assertAlmostEqual(b, b1, delta=1e-8)
 
+    # regression for this issue: https://github.com/huggingface/transformers/issues/12970
+    @require_torch_non_multi_gpu
```
Comment (sgugger) [inline tests/test_trainer.py:831]:
```suggestion
```

File: tests/test_trainer.py [hunk 3]
```diff
@@ -827,6 +827,26 @@ def test_resume_training_with_randomness(self):
...
+    def test_training_with_resume_from_checkpoint_flase(self):
+        # This test will fail flakily for more than 1 GPUs since the result will be slightly more different
+        # TODO: investigate why it fails for 2 GPUs?
+
+        if torch.cuda.is_available():
+            torch.backends.cudnn.deterministic = True
```
Comment (sgugger) [inline tests/test_trainer.py:837]:
```suggestion
```

There are no assert so this is not necessary.

File: tests/test_trainer.py [hunk 4]
```diff
@@ -827,6 +827,26 @@ def test_resume_training_with_randomness(self):
         self.assertAlmostEqual(a, a1, delta=1e-8)
         self.assertAlmostEqual(b, b1, delta=1e-8)
 
+    # regression for this issue: https://github.com/huggingface/transformers/issues/12970
+    @require_torch_non_multi_gpu
+    def test_training_with_resume_from_checkpoint_flase(self):
+        # This test will fail flakily for more than 1 GPUs since the result will be slightly more different
+        # TODO: investigate why it fails for 2 GPUs?
```
Comment (sgugger) [inline tests/test_trainer.py:834]:
```suggestion
```

There are no asserts, so this is not needed.

**Review by sgugger — APPROVED** [review]
Thanks for adding a test as well!

File: src/transformers/trainer.py (hunk 1 above)
Comment (PhilipMay) [reply src/transformers/trainer.py:1008]: removed this

**Comment by sgugger** [comment]
Thanks again!

**Pull request merged**